Code/Framework/AzToolsFramework/Tests/ActionManager/MenuManagerTests.cpp

```diff
@@ -793,4 +811,98 @@ namespace UnitTest
         EXPECT_EQ(menu->actions().size(), 1);
     }
 
+    TEST_F(ActionManagerFixture, VerifySubMenuIsHiddenWhenEmpty)
```

Comment:
comment: Test looks great and the comments are super helpful but it might be worth keep expectations only at the end rather than having some sprinkled throughout because there are then multiple reasons for the test to fail. It may potentially be worth splitting the test for this reason (this could be a follow-up change though)

Reply:
I have split it in two, I could remove the checks in the middle since they're already tested elsewhere but it makes more sense to leave them to me :)

Reply:
Okay sure thing, thanks for the info 👍